src/options.ts

```diff
@@ -269,30 +306,23 @@ const getOptions = () => {
     micromambaRootPath: parseOrUndefined('micromamba-root-path', z.string()),
     micromambaBinPath: parseOrUndefined('micromamba-binary-path', z.string())
   }
-  core.debug(`Inputs: ${JSON.stringify(inputs)}`)
-  validateInputs(inputs)
-  const options = inferOptions(inputs)
-  core.debug(`Inferred options: ${JSON.stringify(options)}`)
-  checkForKnownIssues(options)
-  assertOptions(options)
-  return options
-}
 
-let _options: Options
-try {
-  _options = getOptions()
-} catch (error) {
-  if (core.isDebug()) {
+  try {
+    core.debug(`Inputs: ${JSON.stringify(inputs)}`)
+    validateInputs(inputs)
+    const options = inferOptions(inputs)
+    core.debug(`Inferred options: ${JSON.stringify(options)}`)
+
+    checkForKnownIssues(options)
+    assertOptions(options)
+
+    return options
+  } catch (error) {
+    if (!core.isDebug()) {
+      const message = error instanceof Error ? error.message : typeof error === 'string' ? error : 'Unknown error'
+      core.setFailed(message)
+      exit(1)
```

Comment:
We can kick the exit and debug message stuff in here and just let it throw the error instead. It will be catched in `main.ts` and `post.ts`